libvuln/updates/manager.go

```diff
@@ -142,11 +142,15 @@ func (m *Manager) Run(ctx context.Context) error {
 	// If construction fails, we will simply ignore those updater
 	// sets.
 	for _, factory := range m.factories {
+		updateTime := time.Now()
 		set, err := factory.UpdaterSet(ctx)
 		if err != nil {
 			zlog.Error(ctx).Err(err).Msg("failed constructing factory, excluding from run")
 			continue
 		}
+		if m.updaterSetUpToDate(ctx, set, updateTime) {
```

Comment:
This is confusing to read, the code reads: 
`if the updater set is up to date then don't add the updater set's updaters to the updaters to be run`
But in fact it's saying:
`if the updater is a stub updater then record the update time but make sure it doesn't add its updaters to the updaters to be run`. Maybe we could discuss a way to make this clearer to the uninitiated eye.

Reply:
Yep very true! I’ve had a go at making it clearer let me know what you think and happy to discuss alternatives. Unfortunately it’s made the already long `run` function longer, which I think is what I was trying to avoid initially, hiding it in another function, which then made it unclear what what actually happening